newrelic/resource_newrelic_cloud_gcp_link_account.go

```diff
@@ -0,0 +1,148 @@
+package newrelic
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/newrelic/newrelic-client-go/pkg/cloud"
+)
+
+func resourceNewRelicCloudGcpLinkAccount() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNewRelicCloudGcpLinkAccountCreate,
+		ReadContext:   resourceNewRelicCloudGcpLinkAccountRead,
+		UpdateContext: resourceNewRelicCloudGcpLinkAccountUpdate,
+		DeleteContext: resourceNewRelicCloudGcpLinkAccountDelete,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "name of the linked account",
+				Required:    true,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "project id of the Gcp account",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceNewRelicCloudGcpLinkAccountCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
```

Comment:
Try to add some spacing/newlines to your code, almost like putting it in paragraphs. Example: https://github.com/newrelic/terraform-provider-newrelic/pull/1608/files#diff-e7f2180b00f9dc0400207dc507066a63d1b515cee7da2daf4acabd59f12d2ca5R54

Here Maria does the following
- First paragraph get config, and create the client
- Second paragraph: Get the accountId
- Third paragraph: Get the link account Input

It makes your code easier to read for other developers. By creating these paragraphs you're creating a structure: first this, then this, then this. If you put everything together it's much harder to see which code belongs together.